examples/typed-state/README.md

```diff
@@ -0,0 +1,263 @@
+
+Note that the reads/writes have to be a subset of the state object. In this case we use the global `ApplicationState` object as described above, although it can use a subset/compatible set of fields (or, if you elect not to use centralized state, it just has to be compatible with upstream/downstream versions).
+
+Under the hood, burr will subset the state class so it only has the relevant fields (the reads/write) fields.
+
+```python
+@action.pydantic(reads=["transcript"], writes=["post"])
+def generate_post(state: ApplicationState, llm_client) -> ApplicationState:
+    """Use the Instructor LLM client to generate `SocialMediaPost` from the YouTube transcript."""
+
+    # read the transcript from state
+    transcript = state.transcript
+
+    response = llm_client.chat.completions.create(
+        model="gpt-4o-mini",
+        response_model=SocialMediaPost,
+        messages=[...]
+    )
+    # mutate in place
+    state.post = response
+    # return the state
+    return state
+```
+
+## Typed state for streaming actions
+
+This code for this is in [application.py](application.py).
+
+For streaming actions, not only do we have to type the input/output state, but we can also type the intermediate result.
+
+In this case, we just use the `SocialMediaPost` as we did in the application state. Instructor will be streaming that in as it gets created.
+
+`@streaming_action.pydantic` currently requires you to pass in all the pydantic models as classes, although we will be adding the option to derive from the function signature.
+
+We first call out to OpenAI, then we stream through
+
+```python
+
+@streaming_action.pydantic(
+    reads=["transcript"],
+    writes=["post"],
+    state_input_type=ApplicationState,
+    state_output_type=ApplicationState,
+    stream_type=SocialMediaPost,
+)
+def generate_post_streaming(
+    state: ApplicationState, llm_client
+) -> Generator[Tuple[SocialMediaPost, Optional[ApplicationState]], None, None]:
+    """Streams a post as it's getting created. This allows for interacting data on the UI side of partial
+    results, using instructor's streaming capabilities for partial responses:
+    https://python.useinstructor.com/concepts/partial/
+
+    :param state: input state -- of the shape `ApplicationState`
+    :param llm_client: the LLM client, we will bind this in the application
+    :yield: a tuple of the post and the state -- state will be non-null when it's done
+    """
+
+    transcript = state.transcript
+    response = llm_client.chat.completions.create_partial(
+        model="gpt-4o-mini",
+        response_model=SocialMediaPost,
+        messages=[...],
+        stream=True,
+    )
+    for post in response:
+        yield post, None
+    state.post = post
+    yield post, state
+```
+
+When we call out to the application we built, we have to do a little magic to get typing to work
+in the IDE, but we still have the same benefits as the non-streaming approach.
```

Comment:
what work are you referring to? Either explain or delete this.

Reply:
Added a type-hint, added explanation